client.go

```diff
@@ -421,6 +424,9 @@ type Client struct {
 	// ErrorHandler specifies the custom error handler to use, if any
 	ErrorHandler ErrorHandler
 
+	// Postprocess can prepare the request for retry operation, for example re-sign it
+	Postprocess Postprocess
```

Comment:
Probably in order to merge this change into original repo, we would need to add some tests for new feature

Reply:
That one is still open to keep in mind. But apart from that it LGTM